src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -963,6 +991,24 @@ private class FactoryContext
             public bool UsingTempSourceString { get; set; }
             public List<ParameterExpression> ExtraLocals { get; } = new();
             public List<Expression> ParamCheckExpressions { get; } = new();
+
+            public Dictionary<string, string> TrackedParameters { get; } = new();
+
+            public bool HasAnotherBodyParameter { get; set; }  
+        }
+
+        private static class RequestDelegateFactoryConstants
+        {
+            public const string RouteAttribue = "Route Attribute";
```

Comment:
I think we can better seperate the source and whether it's inferred visually using parens. E.g.

```suggestion
            public const string RouteAttribue = "Route (Attribute)";
```

Reply:
I don't have any preference as long as the user understands that parameters without explicit attributes are inferred. I hope the docs will do a better job to explain that.

Reply:
I think I prefer the parenthesis as well